### ISSUE TITLE
Can't run linter goanalysis_metalinter: inspect: failed to load package jsonpatch: could not load export data: no export data for \"github.com/evanphx/json-patch/v5\"

### ISSUE THREAD

**Issue by lucastheisen** [golangci/golangci-lint#5051]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [X] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [X] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [X] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Description of the problem

When running as part of a [gitlab action](https://github.com/pastdev/clconf/actions/runs/11132412282/job/30936179345?pr=52), the step fails with:

```console
  Running [/home/runner/golangci-lint-1.61.0-linux-amd64/golangci-lint run] in [/home/runner/work/clconf/clconf] ...
  Received 232188 of 232188 (100.0%), 0.2 MBs/sec
  level=warning msg="[runner] Can't run linter goanalysis_metalinter: inspect: failed to load package jsonpatch: could not load export data: no export data for \"github.com/evanphx/json-patch/v5\""
  level=error msg="Running error: can't run linter goanalysis_metalinter\ninspect: failed to load package jsonpatch: could not load export data: no export data for \"github.com/evanphx/json-patch/v5\""
  
  Error: golangci-lint exit with code 3
  Ran golangci-lint in 6975ms
```

When running local, i cannot reproduce this error:

```console
ltheisen@ltop ~/egit/pastdev-clconf
$ ~/tmp/glcil/golangci-lint-1.61.0-linux-amd64/golangci-lint run --max-same-issues 100000 --max-issues-per-linter 1000000

ltheisen@ltop ~/egit/pastdev-clconf
$ echo $?
0
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# Paste output here
```

I cannot run this command directly as it is part of the gitlab action, but the configuration explicitly says:

```yaml
  lint:
    name: lint
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-go@v5
        with:
          go-version-file: go.mod
          cache: true
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v6
        with:
          version: v1.61
```

</details>

### Configuration

<details>

[.golangci.yml](https://github.com/pastdev/clconf/blob/80c5f72e739f11eea00c8cb6b4398246863a9810/.golangci.yml)

```console
---
linters:
  disable-all: true
  enable:
  - asciicheck
  - bidichk
  - durationcheck
  - exhaustive
  - errcheck
  - errorlint
  - gochecknoinits
  - goconst
  - gocritic
  - gofmt
  - gosec
  - gosimple
  - govet
  - ineffassign
  - nakedret
  - nilerr
  - nolintlint
  - revive
  - sqlclosecheck
  - staticcheck
  - typecheck
  - unparam
  - unused
  - wastedassign
  - whitespace
  - wrapcheck
issues:
  include:
  # Reenable some checks golangci-lint disables by default (see golangci-lint run --help)
  - EXC0001
  - EXC0004
  - EXC0005
  - EXC0006
  - EXC0007
linters-settings:
  errcheck:
    ignore: "" # Disable the overbroad ignore rules (they ignore things like Fprintf)
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

I cannot get this output explicitly because its a github action, but that action config is:

```
  lint:
    name: lint
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-go@v5
        with:
          go-version-file: go.mod
          cache: true
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v6
        with:
          version: v1.61
```

which uses my [go.mod for version](https://github.com/pastdev/clconf/blob/80c5f72e739f11eea00c8cb6b4398246863a9810/go.mod#L25) which is:

```text
go 1.17
```

The env is what gets set by `actions/setup-go@v5`.

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

After adding `-v` to the args in my action config, the output is:

```text
  Running [/home/runner/golangci-lint-1.61.0-linux-amd64/golangci-lint run  -v] in [/home/runner/work/clconf/clconf] ...
  Received 232188 of 232188 (100.0%), 0.2 MBs/sec
  level=info msg="golangci-lint has version 1.61.0 built with go1.23.1 from a1d6c560 on 2024-09-09T17:44:42Z"
  level=info msg="[config_reader] Config search paths: [./ /home/runner/work/clconf/clconf /home/runner/work/clconf /home/runner/work /home/runner /home /]"
  level=info msg="[config_reader] Used config file .golangci.yml"
  level=info msg="[lintersdb] Active 25 linters: [asciicheck bidichk durationcheck errcheck errorlint exhaustive gochecknoinits goconst gocritic gofmt gosec gosimple govet ineffassign nakedret nilerr nolintlint revive sqlclosecheck staticcheck unparam unused wastedassign whitespace wrapcheck]"
  level=info msg="[loader] Go packages loading at mode 575 (deps|types_sizes|compiled_files|exports_file|files|imports|name) took 3.036553838s"
  level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 7.125238ms"
  level=info msg="[linters_context/goanalysis] analyzers took 8.986602725s with top 10 stages: buildir: 6.205361567s, the_only_name: 728.649032ms, exhaustive: 654.855556ms, inspect: 256.964272ms, printf: 158.775463ms, ctrlflow: 129.19702ms, bidichk: 105.347993ms, fact_deprecated: 96.355674ms, fact_purity: 91.083523ms, SA5012: 83.371442ms"
  level=warning msg="[runner] Can't run linter goanalysis_metalinter: inspect: failed to load package jsonpatch: could not load export data: no export data for \"github.com/evanphx/json-patch/v5\""
  level=info msg="[runner] processing took 2.227µs with stages: max_same_issues: 291ns, skip_dirs: 251ns, cgo: 180ns, filename_unadjuster: 171ns, invalid_issue: 161ns, nolint: 161ns, max_from_linter: 151ns, identifier_marker: 150ns, skip_files: 140ns, exclude-rules: 51ns, fixer: 50ns, path_shortener: 50ns, source_code: 50ns, path_prefixer: 50ns, uniq_by_line: 40ns, max_per_file_from_linter: 40ns, diff: 40ns, exclude: 40ns, severity-rules: 40ns, autogenerated_exclude: 40ns, sort_results: 40ns, path_prettifier: 40ns"
  level=info msg="[runner] linters took 3.714493502s with stages: goanalysis_metalinter: 3.714462053s"
  level=error msg="Running error: can't run linter goanalysis_metalinter\ninspect: failed to load package jsonpatch: could not load export data: no export data for \"github.com/evanphx/json-patch/v5\""
  level=info msg="Memory: 69 samples, avg is 138.7MB, max is 325.1MB"
  level=info msg="Execution took 6.762238983s"
  Error: golangci-lint exit with code 3
  Ran golangci-lint in 6863ms
```

</details>

### A minimal reproducible example or link to a public repository

<details>

The public repository is: https://github.com/pastdev/clconf, specifically this PR:

https://github.com/pastdev/clconf/pull/52

```go
// add your code here
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by lucastheisen** [comment]
Well, no sooner did i open this issue then notice [another comment on `golangci-lint-action`](https://github.com/golangci/golangci-lint-action/issues/1107#issuecomment-2376906920) that indicates version of go itself could have something to do with behavior of the linter...  So i updated my action to use go version 1.22 instead of the version my project builds at (1.17) and [the linter passes](https://github.com/pastdev/clconf/actions/runs/11132774854/job/30937391040?pr=52).
